test/404s.test.js

```diff
@@ -1738,6 +1738,41 @@ test('400 in case of bad url (pre find-my-way v2.2.0 was a 404)', t => {
     })
   })
 
+  t.test('No route registered', t => {
+    t.plan(3)
+    const fastify = Fastify()
+    fastify.inject({
+      url: '/%c0',
+      method: 'GET'
+    }, (err, response) => {
+      t.error(err)
+      t.equal(response.statusCode, 400)
+      t.same(JSON.parse(response.payload), {
+        error: 'Bad Request',
+        message: "'%c0' is not a valid url component",
+        statusCode: 400
```

Comment:
Why isn't this a 404? The route for `%c0` hasn't been defined. In fact, `%c0` is completely valid according to https://datatracker.ietf.org/doc/html/rfc1738#section-2.2

Reply:
It is possible to throw `404` if we use change `onBadUrl` to `basic404`. Should we change it to `404` according to IETF @mcollina?

Reply:
It is possible to throw `404` if we use `basic404` for the `onBadUrl`. Should we change it to throwing 404 according to IETF @mcollina ?

Reply:
I would go with a 404.